### PR TITLE
Param validation

### LINKED ISSUE #263: PR 261 Bugs
## Description
- PR #261 did not address all of the PR feedback
- there are still typos and spacing issues in the document
- new issue: the parameter should be "q", not "query"

## Expected Behavior
- A clear and concise description of what you expected to happen.

### PR DESCRIPTION
**Pull request by jyotsnaravikumar**
## Type of PR

- [ X] Documentation changes
- [ ] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## Purpose of PR
Provide bug fixes 

## Issues Closed or Referenced

- Closes #263, #243

### CODE REVIEW

File: docs/ParameterValidation.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+- [Refer to the rules for Query Parameters for Movies](###movies-query-parameters)
+
+### Actors API Parameters
+
+- API /api/actors/{actorId}
+- Name: actorId
+- Type: string
+- Parameter Validation:
+  - Valid input: well-formed actorId rule - starts with 'nm' followed by 5-9 digits,parsed as an int
+    - Example: 'nm1265067','NM123456'
```
Comment (aflinchb) [inline docs/ParameterValidation.md]: This is a different rule than the actorId parameter mentioned earlier.  I think this should require the 'nm' to be lowercase to be valid to match the other id parameters in this doc (actorId and movieId).

File: docs/ParameterValidation.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+- Name: q (search)
+- Description: This is a case insensitive search query (q) string on movie title or actor name field
+- Type: string
+- Parameter Validation:
+  - Valid input: query length >= 3 non-whitespace characters
+    - Example: 'ring','Ring'
+    - Status: 200
+    - Content-Type: application/json
+    - Response Body: JSON Array of movies filtered by search query on title or actor name field as [{},{},..],empty array when no results
+  - Invalid input: query length < 3 characters
```
Comment (aflinchb) [inline docs/ParameterValidation.md]: Small nit change - in line 37 right before "empty array" and a few other places below, there are commas without spaces after.  Not a big deal, but was a little weird to read.  Maybe do one more scrub for spaces after commas and other spacing.

File: docs/ParameterValidation.md (hunk 1 above)
Comment (bartr) [reply docs/ParameterValidation.md]: yes, NM123456 and TT123456 are invalid - sorry I missed that

Comment (jyotsnaravikumar) [reply docs/ParameterValidation.md]: Yes, updated these.

File: docs/ParameterValidation.md (hunk 2 above)
Comment (jyotsnaravikumar) [reply docs/ParameterValidation.md]: updated!

**Pull request merged**